packages/hardware-ledger/src/LedgerKeyAgent.ts

```diff
@@ -615,8 +737,66 @@ export class LedgerKeyAgent extends KeyAgentBase {
     }
   }
 
+  async signCip8Data(request: cip8.Cip8SignDataContext): Promise<Cip30DataSignature> {
+    try {
+      const { signingPath, addressParams } = getDerivationPath(
+        request.signWith,
+        request.knownAddresses,
+        this.accountIndex,
+        this.purpose
+      );
+
+      const messageData: MessageData = {
+        address: addressParams,
+        addressFieldType: MessageAddressFieldType.ADDRESS,
+        hashPayload: false,
+        messageHex: request.payload,
+        network: {
+          networkId: this.chainId.networkId,
+          protocolMagic: this.chainId.networkMagic
+        },
+        preferHexDisplay: false,
+        signingPath
+      };
+
+      const deviceConnection = await LedgerKeyAgent.checkDeviceConnection(
+        this.#communicationType,
+        this.deviceConnection
+      );
+
+      const result = await deviceConnection.signMessage(messageData);
+
+      // Re-create the CIP-008 payload the same way the firmware does it internally, otherwise the signature
```

Comment:
What's going on here? This seems like a lot of work and duplication of responsibility on our side 🤔

Reply:
So the way the firmware works is by building the CIP-08 message internally and signing that, but then it only returns the signature and the key used to sign, CIP-30 signData requires that we return the structure used to compute the signature, so our only option is to build it the same way the ledger does it internally as it doesn't return it

Reply:
Urgh, seems like an oversight in the library.